nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -722,3 +812,74 @@ func resourceNsxtUpgradeRunUpdate(d *schema.ResourceData, m interface{}) error {
 func resourceNsxtUpgradeRunDelete(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
+
+func getHostDefaultUpgradeGroup(m interface{}, hostID string) (string, error) {
+	connector := getPolicyConnector(m)
+	hostClient := nsx.NewTransportNodesClient(connector)
+	host, err := hostClient.Get(hostID)
+	if err != nil {
+		return "", err
+	}
+	converter := bindings.NewTypeConverter()
+	base, errs := converter.ConvertToGolang(host.NodeDeploymentInfo, model.HostNodeBindingType())
+	if errs != nil {
+		return "", errs[0]
+	}
+	node := base.(model.HostNode)
+
+	if node.ComputeCollectionId != nil {
+		return *node.ComputeCollectionId, nil
+	}
+
+	// This host is not a part of a compute cluster:
+	// it should be assigned to the 'Group 1 for ESXI' group (this value is hardcoded in NSX)
+	groupClient := upgrade.NewUpgradeUnitGroupsClient(connector)
+	componentType := "HOST"
+	hostGroups, err := groupClient.List(&componentType, nil, nil, nil, nil, nil, nil, nil)
```

Comment:
It seems wasteful to run this API for each `getHostDefaultUpgradeGroup` call, can the logic be change to run it once? We only care for default groups here, and default groups are not expected to change throughout the update operation, is that correct?

Reply:
You mean, load the entire host/group map and cache it for the entire execution?
As the "original" upgrade group for each host could be different according to cluster membership (or lack of cluster association).

Reply:
Yes, only run the API once and use the list for all calls of this function